### PR TITLE
Update Ruby to v3.2.2

### PR DESCRIPTION
**Pull request by ashie**
* Dependent gems are also updated
* Remove shotgun because it doesn't support Ruby 3
* Use latest version of shinatra-asset-pipeline in GitHub to resolve conflict of required rake version
  * To make sure to exist the specified revision, use our fork repository
* Replace `File.exists?` with `File.exist?`
* Remove jemalloc gem because it doesn't seem used in actual and it uses `File.exists?`

### CODE REVIEW

**Comment by daipom** [comment]
Thanks!!

I have succeeded in running the server locally with Ruby 3.2.2.

```console
$ bundle
$ bundle exec rake server
```

File: config/puma.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 preload_app!
 
-rackup      DefaultRackup
+rackup      DefaultRackup if defined?(DefaultRackup)
```
Comment (daipom) [inline config/puma.rb:7]: Certainly, without this fix, we get the following error.
I don't understand what is happening...

```
config/puma.rb:7:in `_load_from': uninitialized constant Puma::DSL::DefaultRackup (NameError)
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/puma/dsl.rb:132:in `instance_eval'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/puma/dsl.rb:132:in `_load_from'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/puma/configuration.rb:237:in `block in load'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/puma/configuration.rb:237:in `each'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/puma/configuration.rb:237:in `load'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/puma/launcher.rb:54:in `initialize'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/rack/handler/puma.rb:68:in `new'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/puma-6.2.2/lib/rack/handler/puma.rb:68:in `run'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/sinatra-2.2.4/lib/sinatra/base.rb:1587:in `start_server'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/sinatra-2.2.4/lib/sinatra/base.rb:1520:in `run!'
	from /home/daipom/work/fluentd/fluentd-website/vendor/ruby/3.2.0/gems/sinatra-2.2.4/lib/sinatra/main.rb:45:in `block in <module:Sinatra>'
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ jobs:
       CIRCLE_ARTIFACTS: /tmp/circleci-artifacts
       CIRCLE_TEST_REPORTS: /tmp/circleci-test-results
     docker:
-      - image: circleci/ruby:2.7.5
+      - image: cimg/ruby:3.2.2
```
Comment (daipom) [inline .circleci/config.yml:11]: Note for me:

https://circleci.com/docs/next-gen-migration-guide/

File: config/puma.rb (hunk 1 above)
Comment (ashie) [reply config/puma.rb:7]: This configuration is ducumented by Heroku and it seems be like the committed code: https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#config

`DefaultRackup` was removed in puma 6.0: https://github.com/puma/puma/pull/2928

**Comment by ashie** [comment]
The shotguns issue can be resolved by https://github.com/rtomayko/shotgun/pull/74
But we don't need to continue using unmaintained gem since we use it only for development.

**Comment by ashie** [comment]
I confirmed it on my local environment & staging site.

File: config/puma.rb (hunk 1 above)
Comment (daipom) [reply config/puma.rb:7]: I see! Thanks!

**Review by daipom — APPROVED** [review]
LGTM!

**Pull request merged**